src/main/java/org/jenkinsci/plugins/gitclient/verifier/AcceptFirstConnectionVerifier.java

```diff
@@ -18,9 +19,28 @@ public class AcceptFirstConnectionVerifier extends HostKeyVerifierFactory {
 
     @Override
     public AbstractCliGitHostKeyVerifier forCliGit(TaskListener listener) {
-        return tempKnownHosts -> {
-            listener.getLogger().println("Verifying host key using known hosts file, will automatically accept unseen keys");
-            return "-o StrictHostKeyChecking=accept-new -o HashKnownHosts=yes";
+        return (tempKnownHosts, url) -> {
+            String hostnamePort = url.getPort() > 0 ? url.getHost() + ":" + url.getPort() : url.getHost();
+            listener.getLogger().printf("Verifying host key for %s using %s, will automatically accept unseen keys %n", hostnamePort, getKnownHostsFile().toPath());
+            String strictHostKeyChecking;
+            if (!Files.exists(getKnownHostsFile().toPath())) {
+               strictHostKeyChecking = "no";
+            } else {
+
+                List<String> hostKeyLines = Files.readAllLines(getKnownHostsFile().toPath());
```

Comment:
I am now thinking about how to synchronize reading here and then writing that most likely happens [there](https://github.com/jenkinsci/git-client-plugin/blob/38abd1db42054d461f5933c25e53140684accd63/src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java#L2676)

Reply:
added synchronization in [80f6d57](https://github.com/jenkinsci/git-client-plugin/pull/881/commits/80f6d578ed8bf3b68e4c89cd63a81cf18e52b3a8)